### ISSUE TITLE
Bot transferring wrong Pokemon

### ISSUE THREAD

**Issue by Osheeta** [OpenPoGo/OpenPoGoBot#294]
### Actual Behavior

```
[2016-08-12 18:43:30] [Transfer] Transferring all Pokemon with CP NOT above 1600 and IV NOT above 0.6 and excluding [].
[2016-08-12 18:43:30] [Transfer] Transferring Magikarp (#129) with CP 65 and IV 0.89 (1/25)
[2016-08-12 18:43:31] [x] Exiting PokemonGo Bot
```
### Steps to Reproduce

Commit dc5aeceeabcdddc4514c9bdef6478bc9be461bbd seems to be guilty.

**Comment by Osheeta** [comment]
When I use the default transfer_pokemon.json, I get the following (and still wrong) result:

```
[2016-08-12 17:47:17] [Transfer] Transferring all Pokemon according to rules and excluding [].
[2016-08-12 17:47:17] [Transfer] Transferring Magikarp (#129) with CP 26 and IV 0.91 (1/23)
```

**Comment by plgonzalezrx8** [comment]
I don't see the error you are talking about. It says that it will transfer ALL Pokemon with CP NOT above 1600 and IV NOT above  0.89. Keyword "AND", your magikarp doesn't meet BOTH criteria, the since the Magicarp is NOT above 1600, it will be transfered.

Can you also post your config? You also didn't put what is the expected behavior.

**Comment by plgonzalezrx8** [comment]
Also, have you tried to use the -iv and -cp flags?. You need to post more information man so we can help you.

I'm having not problems currently.

**Comment by plgonzalezrx8** [comment]
Can you show us your transferpokemon.json and your config.json files? Those come with default values (examples) but its up to you to config them properly.

**Comment by Osheeta** [comment]
Hum I think you misread, the rule says `and IV NOT above 0.6`, my Magikarp has an IV of 0.89 and should not be transferred.

The  transfer_pokemon.json is the one given as an example (transfer_pokemon.json.example), where the rule is `"Magikarp": { "release_below_cp": 91, "release_below_iv": 0.8, "logic": "and" }`. Again, my Magikarp has an IV of 0.91 and should not be transferred.

It seems the bot want to keep only one Pokemon per family, the one who maximises `combat_power * potential`, whatever the rules are.

**Comment by plgonzalezrx8** [comment]
I didn't misread. You don't understand ho that instruction works. Your Pokémon needs to be above 1600CP AND above .6 IV, both need to be met, you need to lower the CP values, or change the AND for an OR. That way it will keep Pokémon above 1600 OR .6 IV

**Comment by plgonzalezrx8** [comment]
Your Magikarp have .89IV but it doesn't have 1600+ IV therefore both criteria are not met, and that's why it gets transferred.

**Comment by plgonzalezrx8** [comment]
do you mind posting a copy of your current OpenPoGoBot folder? Remove personal information.

Also, when you start the bot, is the transfer Pokémon plugin loaded?

**Comment by plgonzalezrx8** [comment]
I can tell you, the problem is not the list, it's working fine for me on all my instances.

**Comment by plgonzalezrx8** [comment]
You are right, I completely misunderstood.  I'll do some dubugging and see what I can find.

**Comment by anon1** [comment]
The bug is in the file `plugins/transfer_pokemon/__init__.py`

These lines should be changed:
```
                    if rules_logic == 'and' and (within_cp and within_potential):
                        continue
                    elif rules_logic == 'or' and (within_cp or within_potential):
                        continue
```
To this:
```
                    if rules_logic == 'and' and not (not within_cp and not within_potential):
                        continue
                    elif rules_logic == 'or' and not (not within_cp or not within_potential):
                        continue
```

**Comment by wchill** [comment]
^ That bugfix is not really a good one. The problem here is that the wording of the logic makes it kind of annoying to reason about.

I'll push a fix soon.

**Comment by Osheeta** [comment]
Thanks a lot @anon1, your patch solves the problem!

**Comment by MaxLeiter** [comment]
Fixed in #298
